addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2739,6 +2739,194 @@ options.registry.ReplaceImage = options.Class.extend({
     },
 });
 
+options.registry.ConditionalVisibility = options.Class.extend({
+
+    /**
+     * We clear out the current style tag to make everything visible
+     * @override
+     */
+    init: function () {
+        const style = document.getElementById('wrap') && document.getElementById('wrap').getElementsByTagName('style')[0];
+        if (style) {
+            style.parentNode.removeChild(style);
+        }
+        this._super(...arguments);
+    },
+
+    /**
+     * @override
+     */
+    willStart: async function () {
+        await this._super(...arguments);
+        this._userValueWidgets.forEach(async widget => {
+            let ids = this.$target[0].dataset[widget._methodsParams.attributeName + 'Id'];
+            if (!ids || !widget.m2oModel) {
+                widget._value = "[]";
+                return;
+            }
+            ids = ids.split(',').map(id => parseInt(id));
+            const selectedRecords = await this._rpc({
+                model: widget.m2oModel,
+                method: 'read',
+                args: [ids, ['display_name']],
+            });
+            widget._value = JSON.stringify(selectedRecords);
+
+            const validIds = selectedRecords.map(record => record.id);
+            const rottenIds = ids.filter(id => !validIds.includes(id));
+            if (rottenIds) {
+                this.selectValue(false, widget._value, widget._methodsParams);
+            }
+        });
+    },
+
+    /**
+     * @see this.selectClass
+     */
+    selectValue(preivewMode, widgetValue, params) {
+        const recordData = JSON.parse(widgetValue);
+        const ids = [];
+        const codes = [];
+        for (const record of recordData) {
+            ids.push(record.id);
+            codes.push(record[params.callWith]);
+        }
+        if (ids.length && codes.length) {
+            this.$target[0].dataset[params.attributeName + 'Id'] = ids.join(',');
+            this.$target[0].dataset[params.attributeName] = codes.join(',');
+        } else {
+            delete this.$target[0].dataset[params.attributeName];
+            delete this.$target[0].dataset[params.attributeName + 'Id'];
+        }
+    },
+
+    /**
+     * @override
+     */
+    selectDataAttribute(preivewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility' && widgetValue === 'conditional') {
+            const collapse = this.$el.children('we-collapse')[0];
+            collapse.classList.add('active');
+            $(collapse).children('.o_we_collapse_toggler')[0].classList.add('active');
```

Comment:
There is code in charge to handle the opening of collapse elements, you should not duplicate its logic.